tests/test_exporter.py

```diff
@@ -107,6 +107,139 @@ def set_package_requires(poetry: Poetry, skip: set[str] | None = None) -> None:
     poetry._package = package
 
 
+def test_exporter_keeps_extras_for_requirements_txt(
+    tmp_dir: str, poetry: Poetry
+) -> None:
+    poetry.locker.mock_lock_data(  # type: ignore[attr-defined]
```

Comment:
Please make it into a fixture as you are using it in 2 tests already.

Reply:
Hopefully this still works for you, but I instead parameterized the test. This approach seemed more common to me after looking at the other tests in this file.